### PR TITLE
fix status code to obey rfc2616

### PR DESCRIPTION
**Pull request by Rongronggg9**
https://datatracker.ietf.org/doc/html/rfc2616

### CODE REVIEW

**Comment by netnr** [comment]
What you are modifying is the status code of the main body content, I can change "code" to "no" or delete it

**Comment by Rongronggg9** [comment]
It is a pity that you have not carefully read my modifications. I indeed modified the `code` in the response body, but in the meantime, I modified `outStaus` too, which is the status code in the response headers. Modifying the `code` in the response body is just a "plus", not my main purpose. Obeying RFC2616 is important to make sure downstream services (if any) will not deal with the response improperly. For example, originally, when an internal error occurs, it just responds a 200 status code and a JSON containing the error info, which can make downstream service deal it as a valid response (but it is not).

File: cors/index.js [hunk 1]
```diff
@@ -34,20 +34,23 @@ async function handleRequest(event) {
...
             //输出提示
+            const invalid = !(request.method == "OPTIONS" || url.length === 0)
             outBody = JSON.stringify({
-                code: 0,
+                code: invalid ? 400 : 0,
                 usage: 'Host/{URL}',
                 source: 'https://github.com/netnr/workers'
             });
             outCt = "application/json";
+            outStatus = invalid ? 400 : 200;
```
Comment (Rongronggg9) [inline cors/index.js:44]: Here is status code I modified.

File: cors/index.js [hunk 2]
```diff
@@ -34,20 +34,23 @@ async function handleRequest(event) {
...
         }
         //阻断
         else if (blocker.check(url)) {
             outBody = JSON.stringify({
-                code: 415,
+                code: 403,
                 msg: 'The keyword: ' + blocker.keys.join(' , ') + ' was blocklisted by the operator of this proxy.'
             });
             outCt = "application/json";
+            outStatus = 403;
```
Comment (Rongronggg9) [inline cors/index.js:53]: Here is status code I modified.

File: cors/index.js [hunk 3]
```diff
@@ -93,6 +96,7 @@ async function handleRequest(event) {
             code: -1,
             msg: JSON.stringify(err.stack) || err
         });
+        outStatus = 500;
```
Comment (Rongronggg9) [inline cors/index.js:99]: Here is status code I modified.

File: pages/_worker.js [hunk 4]
```diff
@@ -25,20 +25,23 @@ async function handleRequest(request) {
...
             //输出提示
+            const invalid = !(request.method == "OPTIONS" || url.length === 0)
             outBody = JSON.stringify({
-                code: 0,
+                code: invalid ? 400 : 0,
                 usage: 'Host/{URL}',
                 source: 'https://github.com/netnr/workers'
             });
             outCt = "application/json";
+            outStatus = invalid ? 400 : 200;
```
Comment (Rongronggg9) [inline pages/_worker.js:35]: Here is status code I modified.

File: pages/_worker.js [hunk 5]
```diff
@@ -25,20 +25,23 @@ async function handleRequest(request) {
...
         }
         //阻断
         else if (blocker.check(url)) {
             outBody = JSON.stringify({
-                code: 415,
+                code: 403,
                 msg: 'The keyword "' + blocker.keys.join(' , ') + '" was blacklisted by the operator of this proxy.'
             });
             outCt = "application/json";
+            outStatus = 403;
```
Comment (Rongronggg9) [inline pages/_worker.js:44]: Here is status code I modified.

File: pages/_worker.js [hunk 6]
```diff
@@ -84,6 +87,7 @@ async function handleRequest(request) {
             code: -1,
             msg: JSON.stringify(err.stack) || err
         });
+        outStatus = 500;
```
Comment (Rongronggg9) [inline pages/_worker.js:90]: Here is status code I modified.

**Review by netnr — COMMENTED** [review]
accept

**Comment by Rongronggg9** [comment]
Thanks!

**Pull request merged**